early-dev/app.R

```diff
@@ -187,181 +188,164 @@ fact_vars_asl <- names(Filter(isTRUE, sapply(ADSL, is.factor)))
-  code = 'ADTRWF <- ADTR %>%
-            dplyr::filter(AVISIT != "BASELINE")
-          teal.data::col_labels(ADTRWF) <- teal.data::col_labels(ADTR, fill = FALSE)',
-  vars = list(ADTR = adtr)
-)
-
-adrs <- cdisc_dataset(
-  dataname = "ADRS",
-  x = ADRS,
-  code = 'ADRS <- synthetic_cdisc_data("latest")$adrs
-          adrs_labels <- teal.data::col_labels(ADRS, fill = FALSE)
-          ADRS <- ADRS %>%
-            filter(PARAMCD %in% c("BESRSPI", "INVET"))  %>%
-            mutate(ADT = as.Date(ADTM)) %>%
-            droplevels()
-        teal.data::col_labels(ADRS) <- c(adrs_labels, "Analysis Date")',
-  vars = list(ADSL = adsl)
-)
-
-adrsswim <- cdisc_dataset(
-  dataname = "ADRSSWIM",
-  x = ADRSSWIM,
-  keys = c("STUDYID", "USUBJID", "PARAMCD", "AVISIT"),
-  code = 'ADRSSWIM <- synthetic_cdisc_data("latest")$adrs %>%
-            dplyr::filter(PARAMCD == "OVRINV") %>%
-            arrange(USUBJID)',
-  vars = list(ADSL = adsl)
-)
-
-adlb <- cdisc_dataset(
-  dataname = "ADLB",
-  x = ADLB,
-  code = 'ADLB <- synthetic_cdisc_data("latest")$adlb
-          ADLB <- ADLB %>%
-            mutate(
-              ADT = as.Date(ADTM),
-              LBSTRESN = as.numeric(LBSTRESC)
-            ) %>%
-            teal.data::col_relabel(
-              ADT = "Analysis Date",
-              LBSTRESN = "Numeric Result/Finding in Standard Units"
-            )',
-  vars = list(ADSL = adsl)
+keys_list <- teal.data:::default_cdisc_keys
+jk <- teal.data::join_keys(
+  teal.data::join_key("ADSL", "ADSL", keys = get_cdisc_keys("ADSL")),
+  teal.data::join_key("ADAE", "ADAE", keys = get_cdisc_keys("ADAE")),
+  teal.data::join_key("ADCM", "ADCM", keys = get_cdisc_keys("ADCM")),
+  teal.data::join_key("ADEX", "ADEX", keys = get_cdisc_keys("ADEX")),
+  teal.data::join_key("ADRS", "ADRS", keys = get_cdisc_keys("ADRS")),
+  teal.data::join_key("ADLB", "ADLB", keys = get_cdisc_keys("ADLB")),
+  teal.data::join_key("ADTR", "ADTR", keys = c("STUDYID", "USUBJID", "PARAMCD", "AVISIT")),
+  teal.data::join_key("ADTRWF", "ADTRWF", keys = c("STUDYID", "USUBJID", "PARAMCD", "AVISIT")),
+  teal.data::join_key("ADRSSWIM", "ADRSSWIM", keys = get_cdisc_keys("ADRS")),
+  teal.data::join_key("ADAE", keys_list[["ADAE"]]$parent, keys = keys_list[["ADAE"]]$foreign),
+  teal.data::join_key("ADCM", keys_list[["ADCM"]]$parent, keys = keys_list[["ADCM"]]$foreign),
+  teal.data::join_key("ADEX", keys_list[["ADEX"]]$parent, keys = keys_list[["ADEX"]]$foreign),
+  teal.data::join_key("ADRS", keys_list[["ADRS"]]$parent, keys = keys_list[["ADRS"]]$foreign),
+  teal.data::join_key("ADLB", keys_list[["ADLB"]]$parent, keys = keys_list[["ADLB"]]$foreign),
+  teal.data::join_key("ADTR", "ADSL", keys = c("STUDYID", "USUBJID", "PARAMCD", "AVISIT")),
+  teal.data::join_key("ADTRWF", "ADSL", keys = c("STUDYID", "USUBJID", "PARAMCD", "AVISIT")),
+  teal.data::join_key("ADRSSWIM", keys_list[["ADRS"]]$parent, keys = keys_list[["ADRS"]]$foreign)
```

Comment:
This needs to be addressed [here](https://github.com/insightsengineering/teal.data/issues/162)